File: simplesearchview/src/main/java/com/ferfalk/simplesearchview/SimpleOnQueryTextListener.java
package com.ferfalk.simplesearchview;

/**
 * @author Fernando A. H. Falkiewicz
 */
public abstract class SimpleOnQueryTextListener implements SimpleSearchView.OnQueryTextListener {
    @Override
    public boolean onQueryTextSubmit(String query) {
        // No action
        return false;
    }

    @Override
    public boolean onQueryTextChange(String newText) {
        // No action
        return false;
    }

    @Override
    public boolean onQueryTextCleared() {
        // No action
        return false;
    }
}

File: simplesearchview/src/main/java/com/ferfalk/simplesearchview/SimpleOnQueryTextListener.kt
package com.ferfalk.simplesearchview

/**
 * @author Fernando A. H. Falkiewicz
 */
abstract class SimpleOnQueryTextListener : SimpleSearchView.OnQueryTextListener {
    override fun onQueryTextSubmit(query: String): Boolean {
        // No action
        return false
    }

    override fun onQueryTextChange(newText: String): Boolean {
        // No action
        return false
    }

    override fun onQueryTextCleared(): Boolean {
        // No action
        return false
    }
}